src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/model/action/ReadWriteActionConfig.kt

```diff
@@ -52,22 +52,14 @@ data class ReadWriteActionConfig(
         @JvmStatic
         @Throws(IOException::class)
         fun parse(xcp: XContentParser, index: Int): ReadWriteActionConfig {
-            var timeout: ActionTimeout? = null
-            var retry: ActionRetry? = null
-
             ensureExpectedToken(Token.START_OBJECT, xcp.currentToken(), xcp::getTokenLocation)
             while (xcp.nextToken() != Token.END_OBJECT) {
```

Comment:
In this case it will give a some generic exception like `expected Token to be END_OBJECT` but with the while loop it will mention certain field is invalid field. Both implementation should work regardless but having the while can show more detailed message about the invalid field.

Reply:
Changed to follow the suggestion. Since we throw `Invalid START_OBJECT` as beginning of the XContentParser, it also make sense to throw `Invalid END_OBJECT` when we expect end object.